app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadService.java

```diff
@@ -457,6 +468,7 @@ private void updateForeground() {
           } catch (Exception e) {
             // Retry on network error
```

Comment:
Note: this catch catches all Exceptions, not just network errors (as this comment indicates). Perhaps it's worth either updating the comment, removing it, or adding more focused catch statements?